src/test/analyzer_test.ts

```diff
@@ -523,4 +549,193 @@ suite('Analyzer', () => {
           testSeed.events.map(e => e.name), ['fired-event', 'data-changed']);
     });
   });
+
+  suite('race conditions and caching', () => {
+    test.skip('editor simulator of imports that import a common dep', async() => {
+      // Here we're simulating a lot of noop-changes to base.html, which has
+      // two imports, which mutually import a common dep. This stresses the
+      // analyzer's caching.
+
+      const wait = () =>
+          new Promise((resolve) => setTimeout(resolve, Math.random() * 30));
```

Comment:
Still don't love these tests, but with the deterministic tests below I'll survive. Could we at least log each `Math.random()` value we use so that we can know how big/small some of these wait times were if/when the test breaks?